modules/api_reference/devportal_api_reference.api.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * @file
+ * Hooks provided by the Devportal API Reference module.
+ */
+
+/**
+ * @addtogroup hooks
+ * @{
+ */
+
+/**
+ * Returns the array of fields of the API Reference content type.
+ *
+ * Takes the values of the fields from the submitted file and returns the
+ * associative array of the fields and the fields' values of the API Reference
+ * content type.
+ *
+ * @param \Drupal\devportal_api_reference\ReferenceInterface $type
+ *   The handler object of the API reference type.
+ * @param object $doc
+ *   The raw structure.
+ * @param \Drupal\file\FileInterface $file
+ *   The uploaded file.
+ * @param \Symfony\Component\HttpFoundation\Request|null $request
+ *   (optional) The HTTP request.
+ *
+ * @return array
+ *   The associative array of the fields and the fields' values of the API
+ *   Reference content type.
+ */
+function hook_devportal_api_reference_fields(\Drupal\devportal_api_reference\ReferenceInterface $type, $doc, FileInterface $file, ?Request $request = NULL) {
```

Comment:
As `use` lines are better to avoid in `.api.php` files, FQCNs are also needed here.

Reply:
I left out $doc because Travis gave this error earlier: "Expected "object" but found "\stdClass" for parameter type." And @mxr576 suggested to leave out the parameter type hint from the function's parameter and only add it in the docblock.